java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightService.java

```diff
@@ -51,30 +54,42 @@
 
   private final BufferAllocator allocator;
   private final FlightProducer producer;
+  private final ServerAuthHandler authHandler;
   private final ExecutorService executors;
 
-  FlightService(BufferAllocator allocator, FlightProducer producer, ExecutorService executors) {
+  FlightService(BufferAllocator allocator, FlightProducer producer, ServerAuthHandler authHandler,
+      ExecutorService executors) {
     this.allocator = allocator;
     this.producer = producer;
+    this.authHandler = authHandler;
     this.executors = new ContextPropagatingExecutorService(executors);
   }
 
-  private FlightContext makeContext(ServerCallStreamObserver<?> responseObserver) {
+  private CallContext makeContext(ServerCallStreamObserver<?> responseObserver) {
+    // Try to get the peer identity from middleware first (using the auth2 interfaces).
     final ContextAdapter contextAdapter = new ContextAdapter();
-    return new FlightContext(contextAdapter.get(AuthConstants.PEER_IDENTITY_KEY), responseObserver::isCancelled);
+    String peerIdentity = contextAdapter.get(AuthConstants.PEER_IDENTITY_KEY);
```

Comment:
Can we use the fully qualified name here, to differentiate between auth and auth2 constants ?
`org.apache.arrow.flight.auth2.AuthConstants`

Reply:
Did this the other way around. Renamed auth2.AuthConstants to auth2.Auth2Constants and remove the FQCN from the auth.AuthConstants reference.